src/TropicalGeometry/curve.jl

```diff
@@ -0,0 +1,67 @@
+###
+# Tropical curves in Oscar
+# ========================
+###
+
+
+
+###
+# 1. Definition
+# -------------
+# M = typeof(min) or typeof(max):
+#   min or max convention, affecting initial ideals
```

Comment:
I wrote this version with `typeof(min)` and `typeof(max)`. Can you point out where or why this is inefficient? It is the same as storing functions in types a la 
```
struct A{FunctionType}
  f::FunctionType
```
which is pretty common and performant julia code if I understand correctly.

Reply:
It was slow to use them as keys in a dictionary, mostly because hashing was slower than that of symbols by an order of magnitude. So this may be irrelevant here! This experience simply made me wary. Just for reference:
```
julia> using BenchmarkTools; T = typeof(sin); @btime hash(T); @btime hash(:sin);
  27.945 ns (1 allocation: 16 bytes)
  1.458 ns (0 allocations: 0 bytes)
```

Reply:
Using them as parameters in parametric types of course is common and should indeed be no issue. Also no problem comparing the functions. Comparing the types of the functions *is* a bit slower, but that should be fine.
```
julia> using BenchmarkTools

julia> @btime min == max; @btime min === max;
  0.833 ns (0 allocations: 0 bytes)
  0.875 ns (0 allocations: 0 bytes)

julia> @btime :min == :max; @btime :min === :max;
  0.875 ns (0 allocations: 0 bytes)
  0.875 ns (0 allocations: 0 bytes)

julia> T = typeof(min); T2=typeof(max) ; @btime T == T2; @btime T === T2;
  55.498 ns (0 allocations: 0 bytes)
  3.625 ns (0 allocations: 0 bytes)

```

Anyway, if @thofma thinks this is fine, I am happy with it :-)

Reply:
Everything with type parameters comparisons will hopefully be compiled away. Some of the things are not written optimally, there should be more `===` and `isa`.